src/core/DY.Core.Trace.Type.fst

```diff
@@ -39,6 +39,10 @@ type principal = string
 
 type state_id = { the_id: nat; }
 
+/// a type macro for timestamps (indices on the trace)
+
+type timestamp = nat
```

Comment:
Arguably this should be hidden by an interface, but it's at least less bad than in the old version as long as we aren't ever directly manipulating it as a `nat`.

Reply:
This code was not introduced by this PR, but simply moved from `Trace.Base` to `Trace.Type` (because now a timestamp appears in the `Corrupt` event).
Indeed previously we have considered abstracting the `timestamp` type, but as discussed in #30 we use the fact that timestamp are `nat`s in the NSL proof.